### PR TITLE
Johanna/5828 fixed manage user forms

### LINKED ISSUE #5828: [Manage users - User information] Forms do not use proper element and error handling needs improvement
## Area or Component

Settings Page -> Manage users Tab -> User information

## Quick Summary

The forms to edit name, edit email are not using the proper html tag to wrap the form and the submission pattern does not show errors on submit.

### How to Replicate
Check the edit name and edit email form and try to submit incorrect/invalid information.

## Action Requested

Change the div element that wraps the form with a form tag.
Implement the pattern to show error on submission (including focus on error).
Stretch use react-hook-forms to implement the error handling.

## Screenshots (optional)
Screenshot 2023-05-17 at 10 05 50 AM
Screenshot 2023-05-17 at 10 05 41 AM

### PR DESCRIPTION
**Pull request by johanna-skylight**
# FRONTEND PULL REQUEST

## Related Issue

- Resolves #5828 

## Changes Proposed

- Refactor the current implementation of the forms to edit user email and user name.
- Fix a current prod bug where the updated email was not presented back to the user.
- Adjust unit testing and clean the act warnings.

## Additional Information
None

## Testing
Code is available in Dev5.

- Verify that both forms handle validation and focus on error as expected.
- Validate that axeGuided test for forms recognizes the form and all the form checks pass.
- Verify that the data updates as expected.

## Screenshots / Demos
Screenshot 2023-06-06 at 3 39 54 PM
Screenshot 2023-06-06 at 3 39 45 PM

### CODE REVIEW

File: frontend/src/app/utils/email.ts [hunk 1]
```diff
@@ -3,3 +3,6 @@ export const emailIsValid = (emailAddress = "") => {
     /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
   return re.test(String(emailAddress).toLowerCase());
 };
+
+export const emailRegex =
+  /^[a-zA-Z0-9_+&*-]+(?:\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}$/;
```
[the remark this answers is not in the document]
Comment (emyl3) [reply frontend/src/app/utils/email.ts:8]: ^non-blocking question 😅

File: frontend/src/app/Settings/Users/EditUserNameModal.tsx [hunk 2]
```diff
@@ -31,82 +35,71 @@ const EditUserNameModal: React.FC<Props> = ({
...
+        value={formCurrentValues.firstName}
         required={true}
-        onChange={(e) => updateFirstName(e.target.value)}
-        onBlur={() => validateFirstName()}
-        validationStatus={firstNameError ? "error" : undefined}
-        errorMessage={firstNameError}
+        validationStatus={errors.firstName?.type ? "error" : undefined}
+        errorMessage={errors.firstName?.message}
+        registrationProps={register("firstName", {
+          required: "first name is required",
```
[the remark this answers is not in the document]
Comment (johanna-skylight) [reply frontend/src/app/Settings/Users/EditUserNameModal.tsx]: Thanks for catching this! it is resolved now.

File: frontend/src/app/utils/email.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (johanna-skylight) [reply frontend/src/app/utils/email.ts:7]: I have modified the code to use the new regex.

**Review by emyl3 — APPROVED** [review]
LGTM! Thank you for changing those! 😸

**Review by BobanL — APPROVED** [review]
Changes look good!

**Pull request merged**